### ISSUE TITLE
Compiling error for darknet with OCV 3.4.1: (cvdef.h) unknown type name ‘namespace’

### ISSUE THREAD

**Issue by spinoza1791** [AlexeyAB/darknet#414]
Makefile OPENCV=1, Platforms tested: Ubuntu 16.04 (updated), Raspberry Pi Stretch 2017-11-29 (updated), darknet version as of 2p CT, 2/28/2018.

NOTE: Cannot reproduce this with OCV **3.4.0** (works fine).  This affects OCV 3.4.1 only.  Compiling with OPENCV=0 succeeds (no issue).

gcc  -DOPENCV `pkg-config --cflags opencv`  -Wall -Wfatal-errors -Ofast -DOPENCV -c ./src/im2col.c -o obj/im2col.o
In file included from /usr/local/include/opencv2/core/types_c.h:59:0,
                 from /usr/local/include/opencv2/core/core_c.h:48,
                 from /usr/local/include/opencv2/highgui/highgui_c.h:45,
                 from ./src/image.c:14:
/usr/local/include/opencv2/core/cvdef.h:485:1: error: unknown type name ‘namespace’
 namespace cv {
 ^
compilation terminated due to -Wfatal-errors.
Makefile:102: recipe for target 'obj/image.o' failed
make: *** [obj/image.o] Error 1
make: *** Waiting for unfinished jobs....
./src/http_stream.cpp: In member function ‘bool MJPGWriter::write(const cv::Mat&)’:
./src/http_stream.cpp:176:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (n < outlen)
           ^

**Comment by AlexeyAB** [comment]
@spinoza1791 Yes, I get an error on Windows too for OpenCV 3.4.1, so this is bug of C API in the OpenCV 3.4.1.

When I try to `#include "opencv2/core/core_c.h"` I get an error:

> C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\cstdint(18): error C2054: expected '(' to follow 'using'

**Comment by AlexeyAB** [comment]
This is a bug in OpenCV 3.4.1 C API: https://github.com/opencv/opencv/issues/10969

**Comment by achalshah20** [comment]
I am also facing the same issue. Any workaround for this until opencv fixes their c API? Thanks!

**Comment by AlexeyAB** [comment]
@achalshah20 You can try to do this changes, as described here: https://github.com/opencv/opencv/issues/10963#issue-301037811
```diff
--- opencv-3.4.1/modules/core/include/opencv2/core/cvdef.h.orig	2018-02-28 10:16:45.000320632 +0000
+++ opencv-3.4.1/modules/core/include/opencv2/core/cvdef.h	2018-02-28 10:34:30.852346927 +0000
@@ -454,6 +454,7 @@ Cv64suf;
 
 
 // Integer types portatibility
+#if defined __cplusplus
 #ifdef OPENCV_STDINT_HEADER
 #include OPENCV_STDINT_HEADER
 #else
@@ -494,6 +495,9 @@ typedef ::uint64_t uint64_t;
 }
 #endif
 #endif
+#else
+#include <stdint.h>
+#endif
 
 
 //! @}
```

**Comment by insurgent92** [comment]
i chose a simple solution, just use 3.4.0 until the problems is fixed. XD

**Comment by austinmw** [comment]
3.4.0 sample/dnn isn't compatible with Yolo V3 is it?

**Comment by AlexeyAB** [comment]
@austinmw No one version of OpenCV-dnn module doesn't support Yolo v3 at this moment.
